syft/generic/object_storage.py

```diff
@@ -126,3 +133,28 @@ def clear_objects(self, return_self: bool = True):
     def current_objects(self):
         """Returns a copy of the objects in the object storage."""
         return self._objects.copy()
+
+    def find_by_id(self, id):
+        """Local search by id"""
+        if id in self._objects:
+            return self._objects[id]
+        else:
+            return None
+
+    def find_by_tag(self, tag):
+        """Local search by tag
+
+        Args:
+            tag (str): exact tag searched
+
+        Return:
+            A list of results, possibly empty
+        """
+        if tag in self._tag_to_object_ids:
+            results = []
+            for obj_id in self._tag_to_object_ids[tag]:
+                obj = self.find_by_id(obj_id)
+                if obj is not None:
```

Comment:
Why is this check needed? When would we fill the dict with None objects?

Reply:
find_by_id can return None if there is a bug in the coherence of _tag_to_object_ids. It's to improve resilience